### PR TITLE
Add kickstart tests for all user interface modes

### PR DESCRIPTION
**Pull request by poncovka**
Create new tests for the text, graphical, cmdline and vnc mode.

Add a new test type for smoke tests. The smoke tests consists
of two minimal installations. One runs in the text mode and one
runs the graphical mode.

### CODE REVIEW

File: ui_graphical.ks.in [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+%ksappend repos/default.ks
+%ksappend common/common_no_payload.ks
+%ksappend payload/default_packages.ks
+
+# Run the installation in the graphical mode.
+graphical
+
+%post
+# The installation is successful.
+%ksappend validation/success_if_result_empty.ks
```
Comment (jstodola) [inline ui_graphical.ks.in:14]: I think there should be an additional check that the installation was really executed in graphical/text/cmdline mode - for example when there is a problem with X, anaconda fallbacks to text mode:

Starting installer, one moment...
anaconda 33.16.3.26-1.el8 for Red Hat Enterprise Linux 8.3 started.
 * installation log files are stored in /tmp during the installation
 * shell is available on TTY2
 * when reporting a bug add logs from /tmp as separate text/plain attachments
13:16:46 X startup failed, falling back to text mode
================================================================================
================================================================================

Comment (poncovka) [reply ui_graphical.ks.in]: Fixed.

**Comment by poncovka** [comment]
Updated.

**Review by jstodola — APPROVED** [review]
It looks good to me, thank you.

**Review by jkonecny12 — APPROVED** [review]
Looks great to me now. Thanks for these!

**Pull request merged**